management-interface.go

```diff
@@ -64,6 +68,37 @@ func init() {
 
 }
 
+type NetworkConfig struct {
+	Online bool `yaml:"online"`
+}
+
+func WriteNetworkConfig(filepath string, config NetworkConfig) error {
+	outBuf, _ := yaml.Marshal(config)
+	return ioutil.WriteFile(filepath, outBuf, 0640)
+}
+
+func ParseNetworkConfig(filepath string) NetworkConfig {
+
+	// Create a default config
+	config := NetworkConfig{
+		Online: true,
+	}
+
+	inBuf, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		// Create config file
+		outBuf, _ := yaml.Marshal(config)
```

Comment:
Same here

Reply:
Menno, I can't see how to check for (and handle) these errors without ending up with nested error checks.  The ParseConfigFile doesn't return an error and is called from a few places.  I like how if an error occurs, it doesn't fail, it just returns a default config struct.  I want to keep this.

And so if I check for every error, I get code like this:

        inBuf, err := ioutil.ReadFile(filepath)
	if err != nil {
		// Create config file
		outBuf, err := yaml.Marshal(config)
		if err != nil {
			// Log an error message?
                       return config
		}
                ioutil.WriteFile(filepath, outBuf, 0640)
                if err != nil {
			// Log an error message?
                       return config
		}

		return config
	}

This seems to be partially created by the code doing something if an error occurs i.e creating a file if it's not there.  And it's kind of assumed that if ReadFile fails, it's because the file is not there, which I don't really like.

I could use my fileExists function to check for this... and then branch on that, check for errors etc.  it all got more convoluted than it needed to be, so I have left it as is for now.

I'd be interested to know what you'd do here.